### PR TITLE
[docs] Update READMEs with more links to the official docs

### PR DESCRIPTION
**Pull request by Drarig29**
### What and why?

The description was outdated (thank you @andrea-mosk) and the entrypoint README was really big because of the contributing section.

> [!NOTE]
> Can be reviewed commit by commit.

### How?

- Move the contributing section to its own `CONTRIBUTING.md` file.
  - Replaced by a `Development` section, [as in the dd-trace](https://github.com/DataDog/dd-trace-js/tree/master?tab=readme-ov-file#development) repo.
- Update the main description.
- Add all commands in the `Usage` section, including 1 link to the corresponding README and 1 link to the corresponding `docs.datadoghq.com` page.
- Align another READMEs. (e.g. adding `Further reading` sections)

### Validation

Go [here](https://github.com/DataDog/datadog-ci/tree/corentin.girard/update-readme) to see the updated README on this branch.

All command READMEs now have a `Further reading` section with at least one link to the official documentation website, except for 1 command:
```bash
$ grep -L "Further reading" src/commands/**/README.md

src/commands/lambda/README.md
```

because it's single-sourced already, which means that [the corresponding official docs page](https://docs.datadoghq.com/serverless/libraries_integrations/cli/) uses the README as source code.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,153 +39,47 @@ For more ways to install the CLI, see [this section](#more-ways-to-install-the-c
...
+  - `upload`: Upload [Git metadata](src/commands/git-metadata). See [documentation](https://docs.datadoghq.com/integrations/guide/source-code-integration/).
+- `junit`:
+  - `upload`: Upload [JUnit test reports](src/commands/junit). See [documentation](https://docs.datadoghq.com/tests/setup/junit_xml/).
+- `lambda`:
+  - `flare`: Troubleshoot your issues with Datadog monitoring on your [AWS Lambda functions](src/commands/lambda).
+  - `instrument`: Instrument [AWS Lambda functions](src/commands/lambda).
+  - `uninstrument`: Uninstrument [AWS Lambda functions](src/commands/lambda).
+- `metric`: Add [metrics](src/commands/metric) to a CI Visibility Pipeline trace or Job span in Datadog. See [documentation](https://docs.datadoghq.com/continuous_integration/pipelines/custom_tags_and_metrics/).
+- `react-native`:
+  - `xcode`: Upload [React Native sourcemaps](src/commands/react-native) for Error Tracking. See [documentation](https://docs.datadoghq.com/real_user_monitoring/error_tracking/reactnative/).
```
Comment (louiszawadzki) [inline README.md:60]:
```suggestion
  - `codepush`: Upload [React Native CodePush sourcemaps](src/commands/react-native) for Error Tracking. See [documentation](https://docs.datadoghq.com/real_user_monitoring/mobile_and_tv_monitoring/setup/codepush/).
  - `upload`: Upload [React Native sourcemaps](src/commands/react-native) for Error Tracking. See [documentation](https://docs.datadoghq.com/real_user_monitoring/error_tracking/reactnative/).
  - `xcode`: Upload [React Native sourcemaps](src/commands/react-native) for Error Tracking from the XCode bundle build phase. See [documentation](https://docs.datadoghq.com/real_user_monitoring/error_tracking/reactnative/).
```

**Review by louiszawadzki — APPROVED** [review]
All good for RUM, added missing react native commands in the list.

**Review by juli1 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Drarig29) [reply README.md]: Woopsie, forgot them 😅
Thank you!

**Review by andrea-mosk — APPROVED** [review]
LGTM thank you for doing this, it looks much better now

File: src/commands/stepfunctions/uninstrument.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export class UninstrumentStepFunctionsCommand extends Command {
 
   public static usage = Command.Usage({
     category: 'Serverless',
-    description: 'Remove Step Functions log groups subscription filter created by datadog-ci.',
+    description: 'Remove Step Function log groups subscription filter created by datadog-ci.',
```
Comment (duncanista) [inline src/commands/stepfunctions/uninstrument.ts:14]: The product is called `Step Functions`, should we keep it like that?

File: README.md [hunk 3]
```diff
@@ -37,153 +39,51 @@ For more ways to install the CLI, see [this section](#more-ways-to-install-the-c
...
+  - `flare`: Troubleshoot your issues with [Cloud Run service](src/commands/cloud-run) configuration. [🔗](https://docs.datadoghq.com/serverless/google_cloud_run)
+- `dsyms`:
+  - `upload`: Upload [iOS dSYM files](src/commands/dsyms) for Error Tracking (macOS only). [🔗](https://docs.datadoghq.com/real_user_monitoring/error_tracking/ios/)
+- `flutter-symbols`:
+  - `upload`: Upload [Flutter symbols](src/commands/flutter-symbols) for Error Tracking. [🔗](https://docs.datadoghq.com/real_user_monitoring/error_tracking/flutter/)
+- `git-metadata`:
+  - `upload`: Upload [Git metadata](src/commands/git-metadata) for Source Code Integration. [🔗](https://docs.datadoghq.com/integrations/guide/source-code-integration/)
+- `junit`:
+  - `upload`: Upload [JUnit test reports](src/commands/junit) for Test Visibility. [🔗](https://docs.datadoghq.com/tests/setup/junit_xml/)
+- `lambda`:
```
Comment (duncanista) [inline README.md:56]: Nice, thanks a lot for adding the subcommands! 😄

**Review by duncanista — APPROVED** [review]
Looks good on the Serverless side – `lambda`, and `cloud-run`.

File: src/commands/stepfunctions/instrument.ts [hunk 4]
```diff
@@ -33,7 +33,7 @@ export class InstrumentStepFunctionsCommand extends Command {
 
   public static usage = Command.Usage({
     category: 'Serverless',
-    description: 'Subscribe Step Function Log Groups to a Datadog Forwarder.',
+    description: 'Subscribe Step Function log groups to a Datadog Forwarder.',
```
[the remark this answers is not in the document]
Comment (Drarig29) [reply src/commands/stepfunctions/instrument.ts:36]: Yep, exactly 😁 I'll wait for @kimi-p's answer 👍

File: README.md [hunk 5]
```diff
@@ -37,153 +39,51 @@ For more ways to install the CLI, see [this section](#more-ways-to-install-the-c
...
+- `react-native`:
+  - `codepush`: Upload [React Native CodePush sourcemaps](src/commands/react-native) for Error Tracking. [🔗](https://docs.datadoghq.com/real_user_monitoring/mobile_and_tv_monitoring/setupcodepush/)
+  - `upload`: Upload [React Native sourcemaps](src/commands/react-native) for Error Tracking. [🔗](https://docs.datadoghq.com/real_user_monitoring/error_tracking/reactnative/)
+  - `xcode`: Upload [React Native sourcemaps](src/commands/react-native) for Error Tracking from the XCode bundle build phase. [🔗](https://docs.datadoghq.com/real_user_monitoring/error_trackingreactnative/)
+- `sarif`:
+  - `upload`: Upload [Static Analysis Results Interchange Format (SARIF)](src/commands/sarif) reports to Datadog. [🔗](https://docs.datadoghq.com/static_analysis/)
+- `sourcemaps`:
+  - `upload`: Upload [JavaScript sourcemaps](src/commands/sourcemaps) for Error Tracking. [🔗](https://docs.datadoghq.com/real_user_monitoring/browser/)
+- `stepfunctions`:
+  - `instrument`: Subscribe [AWS Step Function](src/commands/stepfunctions) log groups to a Datadog Forwarder. [🔗](https://docs.datadoghq.com/serverless/step_functions/installation/tab=datadogcli)
```
Comment (kimi-p) [inline README.md:70]: - `instrument`: Instrument [AWS Step Function](src/commands/stepfunctions) with Datadog to get logs and traces.

File: README.md [hunk 6]
```diff
@@ -37,153 +39,51 @@ For more ways to install the CLI, see [this section](#more-ways-to-install-the-c
...
+  - `uninstrument`: Unsubscribe a [AWS Step Function](src/commands/stepfunctions) log group from the specified Datadog Forwarder. [🔗](https://docs.datadoghq.com/serverless/step_functions/installation/tab=datadogcli)
```
Comment (kimi-p) [inline README.md:71]: - `uninstrument`: Uninstrument [AWS Step Function](src/commands/stepfunctions).


Thank you for created this PR and update for us, @Drarig29 !

**Review by alai97 — APPROVED** [review]
Looks good for docs!

File: src/commands/sourcemaps/README.md [hunk 7]
```diff
@@ -19,7 +19,7 @@ It is also possible to override the full URL for the intake endpoint by defining
 
 ### `upload`
 
-This command will upload all javascript sourcemaps and their corresponding javascript file to Datadog in order to un-minify front-end stack traces received by Datadog.
+This command will upload all JavaScript sourcemaps and their corresponding JavaScript file to Datadog in order to un-minify front-end stack traces received by Datadog.
```
Comment (BenoitZugmeyer) [inline src/commands/sourcemaps/README.md:22]: **🥜 nitpick:** ‏ This could be interpreted as if we uploaded the source files, but we only upload the bundles. Maybe rephrase it like other sourcemap upload commands (ex: `react-native upload`):

```suggestion
This command will upload all JavaScript sourcemaps and their corresponding JavaScript bundles to Datadog in order to un-minify front-end stack traces received by Datadog.
```

**Review by BenoitZugmeyer — APPROVED** [review]

File: src/commands/sourcemaps/README.md (hunk 7 above)
Comment (Drarig29) [reply src/commands/sourcemaps/README.md:22]: Oh, good nitpick indeed 👍

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+module.exports = [HelloWorldCommand]
+```
+
+Lastly, test files must be created in the `__tests__/` folder. `jest` is used to run the tests and a CI has been set using GitHub Actions to ensure all tests are passing when merging a Pull Request.
+
+The tests can then be launched through the `yarn test` command, it will find all files with a filename ending in `.test.ts` in the repo and execute them.
+
+### Continuous Integration tests
+
+The CI performs tests to avoid regressions by building the project, running unit tests and running one end-to-end test.
```
Comment (juan-fernandez) [inline CONTRIBUTING.md:61]: (nitpick) 
is it just one end-to-end test now? It seems we run e2e for sourcemaps, junit, sarif and sbom too (https://github.com/DataDog/datadog-ci/blob/master/.github/workflows/ci.yml#L86-L112) 

I don't know if we need to dive into details of the synthetic test here: maybe just like to the GHA workflow file and add some comments in there?

**Review by juan-fernandez — APPROVED** [review]
looks great, thanks for doing this. Only 1 nitpick

File: CONTRIBUTING.md (hunk 8 above)
Comment (Drarig29) [reply CONTRIBUTING.md:61]: I agree, this is too much of a deep dive, I'll just remove it as this can be documented internally

**Review by jakepruitt — APPROVED** [review]
Looks good from synthetics end!
